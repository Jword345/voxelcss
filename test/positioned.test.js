'use strict';

const eventListener = require('../lib/eventListener');
const positioned    = require('../lib/positioned');

describe('Positioned module', function() {
    let item;
    let position = [1, 2, 3];
    beforeEach(() => {
        item = positioned(eventListener({}));
    });
    it('can get and set position', () => {
        expect(item.getPosition()).toEqual({x: 0, y: 0, z: 0});
        item.setPosition(position);
        expect(item.getPosition()).toEqual({x: 1, y: 2, z: 3});
    });
    it('can translate in 3-dimensions', () => {
        expect(item.getPosition()).toEqual({x: 0, y: 0, z: 0});
        item.setPosition(position);
        item.translate(1, 2, 3);
        expect(item.getPosition()).toMatchSnapshot();
        item.translate(-2, -4, -6);
        expect(item.getPosition()).toEqual({x: 0, y: 0, z: 0});
    });
    it('can only translate number values', () => {
        expect(item.getPosition()).toEqual({x: 0, y: 0, z: 0});
        item.translate('not', 'a', 'number');
        expect(item.getPosition()).toEqual({x: 0, y: 0, z: 0});
        item.translate(undefined, undefined, undefined);
        expect(item.getPosition()).toEqual({x: 0, y: 0, z: 0});
    });
});
