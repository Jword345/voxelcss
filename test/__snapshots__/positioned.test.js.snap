// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Positioned module can translate in 3-dimensions 1`] = `
Object {
  "x": 2,
  "y": 4,
  "z": 6,
}
`;

exports[`Positioned module can trigger move when setting position 1`] = `
Array [
  Array [
    "move",
  ],
  Array [
    "move",
  ],
  Array [
    "move",
  ],
]
`;
